src/class-parsely.php

```diff
@@ -492,6 +487,37 @@ public function construct_parsely_metadata( array $parsely_options, WP_Post $pos
 		return apply_filters( 'wp_parsely_metadata', $parsely_page, $post, $parsely_options );
 	}
 
+	/**
+	 * Sets all metadata values related to post time.
+	 *
+	 * @param array   $metadata Array containing all metadata. It will be potentially mutated to add keys: dateCreated, dateModified, & datePublished.
+	 * @param WP_Post $post     Post object from which to extract time data.
+	 * @return void
+	 */
+	private function set_metadata_post_times( array &$metadata, WP_Post $post ): void {
+		$date_format = 'Y-m-d\TH:i:s\Z';
+		$post_time   = get_post_time( 'U', true, $post );
+
+		if ( false === $post_time ) {
+			return;
+		}
+
+		$post_time_gmt = gmdate( $date_format, $post_time );
+
+		// Set post created and published time.
+		$metadata['dateCreated']   = $post_time_gmt;
+		$metadata['datePublished'] = $post_time_gmt;
+
+		// Set post modified time.
+		$metadata['dateModified'] = $post_time_gmt;
```

Comment:
Could we set this a inside (for instance) and `else` in the later conditional? We are setting this variable twice, when ideally we should check for the condition and then set it accordingly.

Reply:
Since it's going to be set anyway, it's better practice to avoid `else` clauses - [set it once, and overwrite it if necessary](https://phpmd.org/rules/cleancode.html#elseexpression). It reduces the [cyclomatic complexity](https://www.axelerant.com/blog/reducing-cyclomatic-complexity-and-npath-complexity-steps-for-refactoring).

Reply:
I see your point on cyclomatic complexity and I agree with it. My comment comes from trying to avoid doing useless overwrites, which do waste resources (granted, not many of them).

Reply:
@pauarge can you recheck the function and see if it is OK with you?
I also in general tend to avoid if/else when possible.